src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/AdvisorTests.ps1

```diff
@@ -0,0 +1,405 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+	.SYNOPSIS
+	Tests listing Server advisors
+#>
+function Test-ListServerAdvisors
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$server = SetupServer $rg
+
+	try
+	{
+		$response = Get-AzureRmSqlServerAdvisor -ResourceGroupName $server.ResourceGroupName -ServerName $server.ServerName
+		Assert-NotNull $response
+		Assert-AreEqual 4 $response.Count
+		foreach($advisor in $response)
+		{
+			ValidateServer $advisor $server
+			ValidateAdvisorProperties $advisor
+		}
+	}
+	finally
+	{
+		# Cleanup
+		Remove-ResourceGroupForTest $rg
+	}
+}
+
+<#
+	.SYNOPSIS
+	Tests listing Server advisors with recommended actions
+#>
+function Test-ListServerAdvisorsExpanded
+{
+		# Setup
+	$rg = Create-ResourceGroupForTest
+	$server = SetupServer $rg
+
+	try
+	{
+		$response = Get-AzureRmSqlServerAdvisor -ResourceGroupName $server.ResourceGroupName -ServerName $server.ServerName -ExpandRecommendedActions
+		Assert-NotNull $response
+		Assert-AreEqual 4 $response.Count
+		foreach($advisor in $response)
+		{
+			ValidateServer $advisor $server
+			ValidateAdvisorProperties $advisor
+		}
+	}
+	finally
+	{
+		# Cleanup
+		Remove-ResourceGroupForTest $rg
+	}
+}
+
+<#
+	.SYNOPSIS
+	Tests Getting a Server advisor
+#>
+function Test-GetServerAdvisor
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$server = SetupServer $rg
+
+	try
+	{
+		$response = Get-AzureRmSqlServerAdvisor -ResourceGroupName $server.ResourceGroupName -ServerName $server.ServerName -AdvisorName CreateIndex
```

Comment:
Please fix the spacing in this file, and try to break lines close to 80 columns, you cna use the backtick as a line continuation character for PowerShell scripts